riscv/debug_module.cc

```diff
@@ -580,6 +592,8 @@ bool debug_module_t::dmi_write(unsigned address, uint32_t value)
           if (!dmcontrol.dmactive && get_field(value, DMI_DMCONTROL_DMACTIVE))
             reset();
           dmcontrol.dmactive = get_field(value, DMI_DMCONTROL_DMACTIVE);
+          if (!dmstatus.authenticated)
+            return true;
```

Comment:
I'm lost in this code... what does this return `true` apply to?

Reply:
Do I need to write dmactive before DMAUTHDATA is available? or must I be authenticated before writing DMACTIVE?

Reply:
`return true` here indicates the DMI write was successful. But since the debugger hasn't authenticated yet, we don't allow any of the other bits in dmcontrol to be written.

In this implementation you need to write `dmactive` first, and authentication is reset when `dmactive` is cleared. The spec doesn't specify, and probably should. I'm not sold on these semantics, but they feel reasonable.